### DISCUSSION TITLE
Preview of the Panel component

### DISCUSSION THREAD

**Discussion by johnlindquist** [johnlindquist/kit#95] [Share]
https://user-images.githubusercontent.com/36073/112905085-13bca380-90a7-11eb-9247-7f68638a5cce.mp4

This is releasing late tonight. Here's the code for the above video:

```js
//Menu: Testing Arg Panel
//Description: Testing the sequence
//Author: johnlindquist
//Twitter: @johnlindquist

let showGif = await arg(
  "You can display anything",
  `<div class="flex justify-center">
  <img src="https://media3.giphy.com/media/efCPDKBWq8qfZuYR34/giphy-downsized-medium.gif?cid=0b9ef2f4xptdlgxr2p76xp8gi9hezyptzhbhgggo76e30zcl&rid=giphy-downsized-medium.gif" alt="">
  </div>`
)

let showLinks = await arg(
  "Including links",
  md(`
[Visit scriptkit.app](http://scriptkit.app)

[Join the discussion](http://github.com/johnlindquist/kit/discussions)
`)
)

let term = await arg(
  "This still returns the input",
  md(`## What kind of joke would you like to hear?`)
)

let joke = await arg(
  `Here's a ${term} joke to get your day started:`,
  async () => {
    let response = await get(
      `https://icanhazdadjoke.com/search?term=${term}`,
      {
        headers: {
          Accept: "text/plain",
        },
      }
    )

    return md(
      `${response.data
        .split("\n")
        .filter(string => string?.trim()?.length)
        .map(
          joke => `* ${joke}
`
        )
        .join("")}`
    )
  }
)

let autoUpdate = await arg(
  "Auto-update works too!",
  input => `Have a ${input} day!`
)
```

**Comment by elijahmanor** [comment]
YES! I have several scripts where I would love to start using this! I look forward to playing with the new API. Thanks!

**Reply by johnlindquist** [reply]
@elijahmanor ok, released! Please let me know how the update goes!

**Comment by jsjoeio** [comment]
https://user-images.githubusercontent.com/3806031/113009244-a05d7500-912c-11eb-84fc-f7fe5062637e.mov

Works really nicely! No immediate feedback comes to mind, besides wanting to use the arrow keys to select those links. 
I feel like with many of these scripts I'm writing, I only get ideas after I see what other people have made. 

I do like the `md` helper though and the fact that you can use Markdown with this is huge!

**Reply by johnlindquist** [reply]
Sweet, thanks for testing this for me. It was a _big_ internal refactor which included silently adding `marked` to .kit during the update process. I was confident it would go smoothly, but really happy to see it in action!

Re: Arrows and links. 
Strings -> These are intentionally non-interactive and highly visual
Arrays -> can generate a list for those links.
Objects -> Later this year, objects will be able to generate forms where you can have tons of fun :)

**Comment by elijahmanor** [comment]
Looks like if you have previously used a list with `arg` that those options will show up as you type for the new string version...

![Kapture 2021-03-30 at 10 46 34](https://user-images.githubusercontent.com/86454/113017620-41f0c080-9145-11eb-8c92-b4355afc954c.gif)

```js
// Menu: Preview Panel List Bug
// Description: Bug with Panel
// Author: Elijah Manor
// Twitter: @elijahmanor

let number = await arg('Pick a number', [ 'One', 'Two', 'Three', 'Four', 'Five', 'Six' ]);

let response = await arg(
    'Start typing "Two" to see the bug...',
    `<p>You picked <span class="text-blue-400">${ number }</p>`
)
```

**Reply by johnlindquist** [reply]
Good catch, thanks! I'll release a fix today
